src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -248,36 +255,83 @@ def validate_paths(self, data, schema_name, verrors):
             verrors.add(f"{schema_name}.alldirs", "This option can only be used for datasets")
 
     @private
-    def validate_user_networks(self, other_shares, data, schema_name, verrors):
+    async def resolve_hostnames(self, hostnames):
+        hostnames = list(set(hostnames))
+
+        executor = concurrent.futures.ThreadPoolExecutor(max_workers=8)
+
+        async def resolve(hostname):
+            try:
+                return await self.middleware.loop.run_in_executor(executor, socket.gethostbyname, hostname)
```

Comment:
@themylogin do you think we could benefit from a smaller timeout or is that not necessary?

Reply:
@william-gr having timeout here would be great, however I don't see an easy way to add timeout to individual `socket.gethostbyname` calls

Reply:
@themylogin How about using asyncio timeout in the run_in_executor call? And shutdown(wait=False)

Reply:
@william-gr it will shutdown entire pool. We can create another one then to serve unfinished calls, would this approach be ok to you?

Reply:
How would it shutdown the entire pool?

I am not suggestion to call .shutdown() on timeout.

Reply:
try:
    await asyncio.wait_for(
       self.middleware.loop.run_in_executor(executor, socket.gethostbyname, hostname), 5
    )
except Except:
    ...

would that shutdown the entire pool?

Reply:
Sorry, I misunderstood you. If we do this, 8 (size of the pool) DNS timeouts will prevent subsequent hosts from being resolved.

Reply:
@themylogin Unless we just use run_in_io_thread, any problem with that?
I could think it could be a problem if someone sends thousands of hostnames for resolving but we could limit that.